ci/docker/x86_64-unknown-linux-gnu/Dockerfile

```diff
@@ -1,9 +1,6 @@
+# Source from https://github.com/rust-lang-deprecated/rust-buildbot/blob/master/slaves/dist/Dockerfile
 FROM alexcrichton/rust-slave-dist:2015-10-20b
 USER root
 
 WORKDIR /
-RUN curl https://www.cpan.org/src/5.0/perl-5.28.0.tar.gz | tar xzf -
-WORKDIR /perl-5.28.0
-RUN ./configure.gnu
-RUN make -j$(nproc)
-RUN make install
+RUN curl -L https://github.com/lzutao/rustup.rs/releases/download/perl-5.28.0/perl-5.28.0.tar.gz | tar xzf - -C /
```

Comment:
We should upload it to somewhere rather than in my own fork.
@lzutao

Reply:
Certainly I wouldn't like this part of the change.  I do feel like we can find a better solution where we don't use non-packaged perl at all, but I don't see what it is yet.

Reply:
It's better but not the best solution we have. Should we ping alexcrichton 
to see what he think about this?

Reply:
Yeah... @alexcrichton If you have a moment could you weigh in on a better way to do this part of `rustup` CI?  I really don't like storing built binaries of Perl like this, but I also dislike that we end up rebuilding perl an awful lot.

Reply:
Instead of a prebuilt perl, could this perhaps take the same strategy as https://github.com/rust-lang/rustup.rs/pull/1724 and use the prebuilt images from rust-lang/rust for deployment?

Reply:
Did you mean to use [crosss x86_64 linux dockerfile][1]? Or just build an new
docker image based on alexcrichton/rust-slave-dist:2015-10-20b and upload
it to Docker Hub?

[1]: https://github.com/rust-embedded/cross/blob/master/docker/x86_64-unknown-linux-gnu/Dockerfile

Reply:
No in #1724 it updates to use the exact same images as rust-lang/rust uses for its compilers. The images were mostly used for cross targets like powerpc, but x86_64-unknown-linux-gnu can be moved over as well.

Reply:
I think I'd very much prefer to use the exact same images as `rust-lang/rust` for all our platforms tbf.

Reply:
I tried a bit in my local machine. I think the image size of `rust-x86_64-unknown-linux-gnu` 
from `rust-lang/rust` is rather big (3.13GB) for project likes rustup (with download
time + extracted time, I expect build time will increase). I will measure it in my own fork's
CI.

Reply:
I did the change in [my branch](https://github.com/lzutao/rustup.rs/tree/rlrust).
The [Travis build log](https://travis-ci.com/lzutao/rustup.rs/builds/108571027) here.

There are two blocking points:
* The build is failed, but I don't know why it's failed. It complains about
  ``linker `cc` not found``, but gcc and g++ is included in rust
  x86_64 and i686 docker image?
* The build time is increase. The average build time is 17 minutes and 30 
  seconds (with build succeed and all tests is passed).
  Now the build time is at least 18 minutes (with tests uncompleted).

Reply:
It's possible that travis will cache the image now for us, reducing the cost of using a larger docker image, but the `cc` not found is a pain.

Reply:
@kinnison We use rust's image. This image could be change or updated so the 
cache will be invalid. Correct my if I'm wrong.

Reply:
It would, but not very often.  (I think at least every 90 days but likely more often than that) so I think it won't be too bad.

Reply:
I forgot. We download the tar-compressed docker images from aws with curl (about 1GB). So I don't think Travis would (automatically) cache this file for us. We could force Travis cache this file, but it's not recommended by Travis and is inefficient.